src/main/java/net/earthcomputer/clientcommands/command/ConnectFourCommand.java

```diff
@@ -0,0 +1,363 @@
+package net.earthcomputer.clientcommands.command;
+
+import com.mojang.blaze3d.platform.InputConstants;
+import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.logging.LogUtils;
+import net.earthcomputer.clientcommands.c2c.C2CPacketHandler;
+import net.earthcomputer.clientcommands.c2c.packets.PutConnectFourPieceC2CPacket;
+import net.earthcomputer.clientcommands.features.TwoPlayerGame;
+import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.multiplayer.PlayerInfo;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.HoverEvent;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.resources.ResourceLocation;
+import org.slf4j.Logger;
+
+public class ConnectFourCommand {
+    private static final Logger LOGGER = LogUtils.getLogger();
+
+    public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher) {
+        dispatcher.register(TwoPlayerGame.FOUR_IN_A_ROW_GAME_TYPE.createCommandTree());
+    }
+
+    public static void onPutConnectFourPieceC2CPacket(PutConnectFourPieceC2CPacket packet) {
+        String sender = packet.sender();
+        ConnectFourGame game = TwoPlayerGame.FOUR_IN_A_ROW_GAME_TYPE.getActiveGame(sender);
+        if (game == null) {
+            return;
+        }
+        game.onMove(packet.x(), game.opponentPiece());
+    }
+
+    public static class ConnectFourGame {
+        public static final int WIDTH = 7;
+        public static final int HEIGHT = 6;
+
+        public final PlayerInfo opponent;
+        public final byte yourPiece;
+        public byte activePiece;
+        public final byte[][] board;
+        /**
+         * 0 for undecided
+         * 1 for red
+         * 2 for yellow
+         * 3 for draw
+        */
+        public byte winner;
+
+        public ConnectFourGame(PlayerInfo opponent, byte yourPiece) {
+            this.opponent = opponent;
+            this.yourPiece = yourPiece;
+            this.activePiece = Piece.RED;
+            this.board = new byte[WIDTH][HEIGHT];
+            this.winner = 0;
+        }
+
+        public void onMove(int x, byte piece) {
+            if (piece != activePiece) {
+                LOGGER.warn("Invalid piece, the active piece is {} and the piece that was attempted to be placed was {}", Piece.name(activePiece), Piece.name(piece));
+                return;
+            }
+
+            if (!this.isGameActive()) {
+                LOGGER.warn("Tried to add piece to the already completed game with {}.", this.opponent.getProfile().getName());
+                return;
+            }
```

Comment:
Is this check (and method) necessary? It seems that if the game is finished, the game is removed from `TwoPlayerGame.FOUR_IN_A_ROW_GAME_TYPE.activeGames`. Therefore no code beyond this check can be reached:
```java
ConnectFourGame game = TwoPlayerGame.FOUR_IN_A_ROW_GAME_TYPE.getActiveGame(sender);
if (game == null) {
    return;
}
```

Reply:
the window can remain open when game is complete